File: FlightETS-Java/app/src/main/java/com/example/flightets/FlightAdapter.java
package com.example.flightets;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class FlightAdapter extends RecyclerView.Adapter<FlightAdapter.FlightViewHolder> {

    private List<Flight> flights;

    public FlightAdapter(List<Flight> flights) {
        this.flights = flights;
    }

    public static class FlightViewHolder extends RecyclerView.ViewHolder {
        TextView destinationText, departureText, arrivalText, airlineText, priceText;

        public FlightViewHolder(@NonNull View itemView) {
            super(itemView);
            destinationText = itemView.findViewById(R.id.destinationText);
            departureText = itemView.findViewById(R.id.departureText);
            arrivalText = itemView.findViewById(R.id.arrivalText);
            airlineText = itemView.findViewById(R.id.airlineText);
            priceText = itemView.findViewById(R.id.priceText);
        }
    }

    @NonNull
    @Override
    public FlightViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_flight, parent, false);
        return new FlightViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull FlightViewHolder holder, int position) {
        Flight flight = flights.get(position);
        holder.destinationText.setText("Vers " + flight.getDESTINATION());
        holder.departureText.setText("Départ: " + flight.getHEURE_DEPART());
        holder.arrivalText.setText("Arrivée: " + flight.getHEURE_ARRIVEE());
        holder.airlineText.setText("Compagnie: " + flight.getCOMPAGNIE());
        holder.priceText.setText("Prix: " + flight.getPRIX() + "$");
    }

    @Override
    public int getItemCount() {
        return flights.size();
    }

    // 🔧 Update flight list
    public void setFlights(List<Flight> newFlights) {
        this.flights = newFlights;
        notifyDataSetChanged();
    }
}

File: FlightETS/app/src/main/java/com/example/flightets/FlightAdapter.kt
package com.example.flightets

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class FlightAdapter(private var flights: List<Flight>) :
    RecyclerView.Adapter<FlightAdapter.FlightViewHolder>() {

    class FlightViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val destinationText: TextView = itemView.findViewById(R.id.destinationText)
        val departureText: TextView = itemView.findViewById(R.id.departureText)
        val arrivalText: TextView = itemView.findViewById(R.id.arrivalText)
        val airlineText: TextView = itemView.findViewById(R.id.airlineText)
        val priceText: TextView = itemView.findViewById(R.id.priceText)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FlightViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_flight, parent, false)
        return FlightViewHolder(view)
    }

    override fun onBindViewHolder(holder: FlightViewHolder, position: Int) {
        val flight = flights[position]
        holder.destinationText.text = "Vers ${flight.DESTINATION}"
        holder.departureText.text = "Départ: ${flight.HEURE_DEPART}"
        holder.arrivalText.text = "Arrivée: ${flight.HEURE_ARRIVEE}"
        holder.airlineText.text = "Compagnie: ${flight.COMPAGNIE}"
        holder.priceText.text = "Prix: ${flight.PRIX}$"
    }

    override fun getItemCount() = flights.size

    // 🔧 ADD THIS METHOD
    fun setFlights(newFlights: List<Flight>) {
        flights = newFlights
        notifyDataSetChanged()
    }
}
